src/components/Composer/index.tsx

```diff
@@ -216,16 +218,88 @@ function Composer(
 
             const TEXT_HTML = 'text/html';
 
+            const clipboardDataHtml = event.clipboardData?.getData(TEXT_HTML) ?? '';
+            const clipboardDataTypesHtml = event.clipboardData?.types.includes(TEXT_HTML) ?? false;
+
             // If paste contains files, then trigger file management
             if (event.clipboardData?.files.length && event.clipboardData.files.length > 0) {
                 // Prevent the default so we do not post the file name into the text box
                 onPasteFile(event.clipboardData?.files[0]);
                 return;
             }
 
+            // If clipboardDataHtml contains base64 image source
+            if (clipboardDataHtml?.includes(CONST.IMAGE_BASE64_MATCH)) {
+                const domparser = new DOMParser();
+                const pastedHTML = clipboardDataHtml;
+                const embeddedImages = domparser.parseFromString(pastedHTML, TEXT_HTML)?.images;
+
+                if (embeddedImages.length > 0 && embeddedImages[0].src) {
+                    const src = embeddedImages[0].src;
+
+                    // convert base64 to file blob
+                    const base64ImageContent = src.split(',')[1];
+                    const mimeType = src.split(';')[0].split(':')[1];
+                    const byteString = atob(base64ImageContent);
+                    const arrayBuffer = new ArrayBuffer(byteString.length);
+                    const uint8Array = new Uint8Array(arrayBuffer);
+                    for (let i = 0; i < byteString.length; i++) {
+                        uint8Array[i] = byteString.charCodeAt(i);
+                    }
+                    const blob = new Blob([arrayBuffer], {type: mimeType});
+                    const file = new File([blob], 'image.jpg', {type: 'image/jpeg'});
+
```

Comment:
Can we make use of this available method `base64ToFile`, DRY up the logic here. https://github.com/Expensify/App/blob/82fff8c54517ad2256ca2b7328fc04f538732012/src/libs/fileDownload/FileUtils.ts#L213

Reply:
Correct, I missed that one. Thanks!